remove default enable ACPI kconfig macro such as DSDT and ACPI_SHELL

najumon1980 opened the pull request:
remove default enable kconfig macro such as DSDT and ACPI_SHELL if PCIE or SHELL enabled. All Intel platform may not require to use ACPI DSDT related APIs and hence it should not enable default if PCIE enabled.

najumon1980 committed:
lib: acpica: remove default enable DSDT and ACPI_SHELL

remove default enable kconfig macro such as DSDT and ACPI_SHELL
if PCIE or SHELL enabled

Signed-off-by: Najumon B.A <najumon.ba@intel.com>

najumon1980 committed:
board: x86: adl: enable uart init as post kernel

In Intel Alder Lake platform, by default console UART is exposed as
PCI device and due to ACPI depandancy it need to be post kernel.

Signed-off-by: Najumon B.A <najumon.ba@intel.com>

finikorg commented on boards/x86/intel_adl/intel_adl_crb_defconfig:
```diff
@@ -13,3 +13,4 @@ CONFIG_SMP=y
 CONFIG_BUILD_OUTPUT_EFI=y
 CONFIG_BUILD_NO_GAP_FILL=y
 CONFIG_UART_NS16550_PARENT_INIT_LEVEL=y
+CONFIG_ACPI_DSDT_SUPPORT=y
```
The commit message says about "enable uart init as post kernel "

jhedberg commented:
This PR makes UART work on up_squared, but there are two concerns I have:
- It does this in an indirect way, by disabling `CONFIG_PCIE_PRT`, a Kconfig option that's not mentioned anywhere in the PR or the commit messages. If I understood correctly, it's that option which makes PCIe code depend on ACPI.
- It's not really addressing the main concern, which is that we should work toward getting PCIe and ACPI initialised much earlier in the boot (most likely PRE_KERNEL_1 is the right level)

nashif commented:
> uart driver is started after uart console and after the logger, this will fail.

actually. one uart is starting in the right level, do we have something not configured with pcie?

najumon1980 committed:
lib: acpica: add prt bus name via kconfig

add prt bus name via Kconfig and help to support platform with
diffrent acpi name for PCI bus.

Signed-off-by: Najumon B.A <najumon.ba@intel.com>

najumon1980 commented:
PR heading and content updated

dcpleung commented on lib/acpi/Kconfig:
```diff
@@ -23,7 +23,7 @@ config ACPI_MAX_PRT_ENTRY
 	bool "ACPI command Shell"
 	default y
-	depends on SHELL
+	depends on SHELL && ACPI_DSDT_SUPPORT
```
Please provide some reasons in the commit message on why these need to be done.

jhedberg commented:
@najumon1980 is this PR still relevant? If not, please close it.
